### ISSUE TITLE
Pylint complains about reference to __package__

### ISSUE THREAD

**Issue by jendagroovy** [PyCQA/pylint#1693]
Hi,

### Steps to reproduce
```
# test.py
# Make relative imports work when run as CLI script
if __name__ == '__main__' and __package__ is None:
    __package__ = 'some.package'
```

```
$ python test.py && echo "ok"
ok

$ pylint test.py
No config file found, using default configuration
************* Module test
W:  3, 4: Redefining built-in '__package__' (redefined-builtin)
C:  1, 0: Missing module docstring (missing-docstring)
E:  2,30: Using variable '__package__' before assignment (used-before-assignment)

----------------------------------------------------------------------
Your code has been rated at -25.00/10 (previous run: -25.00/10, +0.00)
```
### Current behavior
```
E:  2,30: Using variable '__package__' before assignment (used-before-assignment)
```

### Expected behavior

As the `__package__` is a built-in, the error should not be reported
### pylint --version output
```
No config file found, using default configuration
pylint 1.7.4,
astroid 1.5.3
Python 2.7.10 (default, Feb  7 2017, 00:08:15)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```
